parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetReader.java

```diff
@@ -114,6 +126,10 @@ private ParquetReader(Configuration conf,
     List<FileStatus> statuses = Arrays.asList(fs.listStatus(file, HiddenFileFilter.INSTANCE));
     List<Footer> footers = ParquetFileReader.readAllFootersInParallelUsingSummaryFiles(conf, statuses, false);
     this.footersIterator = footers.iterator();
+    globalMetaData = ParquetFileWriter.getGlobalMetaData(footers);
+    MessageType schema = globalMetaData.getSchema();
+    Map<String, Set<String>> extraMetadata = globalMetaData.getKeyValueMetaData();
+    readContext = readSupport.init(new InitContext(conf, extraMetadata, schema));
```

Comment:
This is done in the constructor as `readContext` is final and can only be assigned in the constructor. Do we want to make this non-final to make things consistently initialized in `initialize()`?